http-clients/src/main/java/feign/InputStreamDelegateEncoder.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2016 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package feign;
+
+import com.google.common.base.Charsets;
+import feign.codec.EncodeException;
+import feign.codec.Encoder;
+import java.io.IOException;
+import java.io.InputStream;
+import java.lang.reflect.Type;
+
+/**
+ * If the body type is an InputStream, write it into the body, otherwise pass to delegate.
+ */
+public final class InputStreamDelegateEncoder implements Encoder {
+    private final Encoder delegate;
+
+    public InputStreamDelegateEncoder(Encoder delegate) {
+        this.delegate = delegate;
+    }
+
+    @Override
+    public void encode(Object object, Type bodyType, RequestTemplate template) throws EncodeException {
+        if (bodyType.equals(InputStream.class)) {
+            try {
+                template.body(Util.toByteArray((InputStream) object), Charsets.UTF_8);
```

Comment:
also, does this close the InputStream after it's complete? Either way, class javadoc should probably indicate behavior.